pkg/component/extensions/operatingsystemconfig/operatingsystemconfig.go

```diff
@@ -259,14 +259,14 @@ func (o *operatingSystemConfig) reconcile(ctx context.Context, reconcileFn func(
 			return fmt.Errorf("failed reconciling OperatingSystemConfig %s for worker %s: %w", client.ObjectKeyFromObject(osc), worker.Name, err)
 		}
 
-		oscKey, err := o.calculateKeyForVersion(hashVersion, &worker)
+		oscKey, err := o.calculateKey(hashVersion, &worker)
 		if err != nil {
 			return err
 		}
 
 		data := Data{
 			Object:                        osc,
-			IncludeSecretNameInWorkerPool: hashVersion > 1,
+			IncludeSecretNameInWorkerPool: hashVersion > 1 || v1beta1helper.IsUpdateStrategyInPlace(worker.UpdateStrategy),
```

Comment:
Do we need the secret name for in-place worker pools?

Reply:
Since the secret is not supposed to change and it has some metadata for what is configured in the worker pool, I would prefer to keep it for worker pool hash calculation.